internal/hcsoci/resources_lcow.go

```diff
@@ -120,6 +120,19 @@ func allocateLinuxResources(ctx context.Context, coi *createOptionsInternal, r *
 				// Mounts that map to a path in UVM are specified with 'sandbox://' prefix.
 				// example: sandbox:///a/dirInUvm destination:/b/dirInContainer
 				uvmPathForFile = mount.Source
+			} else if strings.HasPrefix(mount.Source, "hugepages://") {
+				// currently we only support 2M hugepage size
+				hugePageSubDirs := strings.Split(strings.TrimPrefix(mount.Source, "hugepages://"), "/")
+				if len(hugePageSubDirs) < 2 {
+					return errors.Errorf(`%s mount path is invalid, expected format: hugepages://<hugepage-size>/<hugepage-src-location>`, mount.Source)
+				}
+
+				// hugepages:// should be followed by pagesize
+				if hugePageSubDirs[0] != "2M" {
+					return errors.Errorf(`Only 2M (megabytes) pagesize is supported, got %s`, hugePageSubDirs[0])
```

Comment:
Capital error message here still, I can fix shortly after as I know this is time critical.